Initial PR

NickLewanowicz opened the pull request:
Server using node.js with express for backend requests of Arctic.

Commit:
First commit: proper routes for post and gets of packages inside packagedb array
Next time: looking to add an actual permanent storage solution

Commit:
Merge branch 'master' of https://github.com/NickLewanowicz/arctic-server

Commit:
File Restructure

Moved bin folder to ./ instead of ./node_modules

Commit:
Dependencies are JSON api compatible. Will work by default with Ember Serializer.

Commit:
- Added developer dependency relationship to repository objects and name to dependency objects.

 - /repositories/:repo/dependencies now displays the repository dependency relationship

Commit:
- Save Developer Dependencies to Database in the dependencies collection.
 - Modified repository model to have a 'devdependencies' relationship

Commit:
Stable release prior to implementing Version Model and removing Dependency Model

Commit:
Large Commit.

Commit:
Small bug fix:
-Previously on startup if you didnt have permission to access a project when populating a database the server would stop population. Now it skips that particular package.json and moves to the next.

Commit:
Stability Improvement
- If user manually enters an invalid url server will return null data (opposed to crash)

NickLewanowicz committed:
Update package.json

NickLewanowicz committed:
Update package.json

Commit:
Minor tweak.

Commit:
Minor modifications

Commit:
Addition of Ecosystems
-Modified the Dependency and Version models to include compliancy properties which represetn the minimum version to get compliance (for depenedcies) and the percent of compliant dependencies (for versions) based on ecosystem.

-Fixed bug were sometimes the server would crash with intial startup due to mongoDB connect call

-Added minRequired collection which will contain documents outlining all required minimum dependencies

Commit:
Data Modification
-Changed dependency to have a boolean to define if its compliant to a particular ecosystem

NickLewanowicz committed:
Readability Improvement

Removed console logs and improved

sglanzer-fire commented:
You're missing a few filters on your .gitignore (pulling in node_modules for example)

NickLewanowicz committed:
Added node_modules to .gitignore

NickLewanowicz committed:
.

NickLewanowicz committed:
.

NickLewanowicz committed:
Semver 'satisfies' bug

Server satisfies would return null if param contained ‘~’ or ‘^’.
Removed those characters before running ‘satisfies’.

NickLewanowicz committed:
Bug fix

#CHANGELOG
- Fixed bug causing versions of repos to not be saved
